### PR TITLE
Switch to npm 7.

### PR DESCRIPTION
**Pull request by mshima**
Fixes https://github.com/jhipster/generator-jhipster/issues/12747.

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: .github/workflows/angular.yml [hunk 1]
```diff
@@ -73,6 +73,7 @@ jobs:
             fail-fast: false
             matrix:
                 node_version: [14.15.0]
```
Comment (DanielFran) [inline .github/workflows/angular.yml:75]: Change to [14.15.x]

File: .github/workflows/angular.yml [hunk 2]
```diff
@@ -73,6 +73,7 @@ jobs:
...
+                npm_version: [7.0.11]
```
Comment (DanielFran) [inline .github/workflows/angular.yml:76]: Change to [7.0.x]

File: .github/workflows/generator.yml [hunk 3]
```diff
@@ -35,11 +35,14 @@ jobs:
             matrix:
                 node_version: [14.15.0]
```
Comment (DanielFran) [inline .github/workflows/generator.yml:36]: Change to [14.15.x]

File: .github/workflows/generator.yml [hunk 4]
```diff
@@ -35,11 +35,14 @@ jobs:
...
                 os: [ubuntu-20.04]
+                npm_version: [7.0.11]
```
Comment (DanielFran) [inline .github/workflows/generator.yml:38]: Change to [7.0.x]

File: .github/workflows/incremental-changelog.yml [hunk 5]
```diff
@@ -70,6 +70,7 @@ jobs:
             fail-fast: false
             matrix:
                 node_version: [14.15.0]
```
Comment (DanielFran) [inline .github/workflows/incremental-changelog.yml:72]: Change to [14.15.x]

File: .github/workflows/incremental-changelog.yml [hunk 6]
```diff
@@ -70,6 +70,7 @@ jobs:
...
+                npm_version: [7.0.11]
```
Comment (DanielFran) [inline .github/workflows/incremental-changelog.yml:73]: Change to [7.0.x]

File: .github/workflows/react.yml [hunk 7]
```diff
@@ -73,6 +73,7 @@ jobs:
             fail-fast: false
             matrix:
                 node_version: [14.15.0]
```
Comment (DanielFran) [inline .github/workflows/react.yml:75]: Change to [14.15.x]

File: .github/workflows/react.yml [hunk 8]
```diff
@@ -73,6 +73,7 @@ jobs:
...
+                npm_version: [7.0.11]
```
Comment (DanielFran) [inline .github/workflows/react.yml:76]: Change to [7.0.x]

File: .github/workflows/vue.yml [hunk 9]
```diff
@@ -73,6 +73,7 @@ jobs:
             fail-fast: false
             matrix:
                 node_version: [14.15.0]
```
Comment (DanielFran) [inline .github/workflows/vue.yml:75]: Change to [14.15.x]

File: .github/workflows/vue.yml [hunk 10]
```diff
@@ -73,6 +73,7 @@ jobs:
...
+                npm_version: [7.0.11]
```
Comment (DanielFran) [inline .github/workflows/vue.yml:76]: Change to [7.0.x]

File: .github/workflows/webflux.yml [hunk 11]
```diff
@@ -72,6 +72,7 @@ jobs:
             fail-fast: false
             matrix:
                 node_version: [14.15.0]
```
Comment (DanielFran) [inline .github/workflows/webflux.yml:74]: Change to [14.15.x]

File: .github/workflows/webflux.yml [hunk 12]
```diff
@@ -72,6 +72,7 @@ jobs:
...
+                npm_version: [7.0.11]
```
Comment (DanielFran) [inline .github/workflows/webflux.yml:75]: Change to [7.0.x]

**Review by DanielFran — APPROVED** [review]
Just few changes to avoid to update workflows with minor changes version of node and npm

**Comment by mshima** [comment]
@DanielFran not sure `npm install npm@7.0.x` works, should be `npm install npm@~7.0.0`.

Shouldn’t we stick with fixed version to match npm used at maven frontend?
https://github.com/jhipster/generator-jhipster/blob/2419625c35382a2c768a9b52e3bd34b9c363dd68/generators/generator-constants.js#L33-L34

**Comment by pascalgrimaud** [comment]
yes, we should use an exact version in the CI, to match with `const NODE_VERSION = '14.15.0'`

**Comment by mshima** [comment]
Looks like most of backward incompatibilities was fixed.

**Comment by mraible** [comment]
I've been using npm 7 for the last 24 hours with JHipster 7 and haven't experienced any issues.

```
$ npm --version
7.4.2
```

**Comment by mshima** [comment]
Problems happens when mixing full build with without `webapp` profile.
That will make node_modules/lock been switched from 6 <-> 7.
A full build installs a npm 6 locally and uses it.

**Comment by pascalgrimaud** [comment]
if it's stable enough, then we could have it for final v7 release

**Comment by mshima** [comment]
@pascalgrimaud yep it's working great IMO, better then npm 6.

**Comment by pascalgrimaud** [comment]
Ok nice @mshima ! I'll test npm7 this Monday

Can you resolve conflicts plz? So this PR can be merged

**Pull request merged**